src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+type Shape = 'rectangle' | 'triangle' | 'circle';
 
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA === 0 || sideB === 0 || sideC === 0) {
+      throw Error('Lengths of the sides must be greater than 0.');
+    }
+
+    if (
```

Comment:
also, let's integrate both and create a separate method like `validateTriangle`  for that purpose